src/commands/edge_app_manifest.rs

```diff
@@ -0,0 +1,549 @@
+        write!(&manifest_file, "---\n{yaml}")?;
+        Ok(())
+    }
+
+    pub fn prepare_payload(manifest: &EdgeAppManifest) -> HashMap<&str, serde_json::Value> {
+        [
+            ("app_id", &manifest.app_id),
+            ("user_version", &manifest.user_version),
+            ("description", &manifest.description),
+            ("icon", &manifest.icon),
+            ("author", &manifest.author),
+            ("homepage_url", &manifest.homepage_url),
+        ]
+        .iter()
+        .filter_map(|(key, value)| value.as_ref().map(|v| (*key, json!(v))))
+        .collect()
+    }
+
+    pub fn ensure_manifest_is_valid(path: &Path) -> Result<(), CommandError> {
+        match EdgeAppManifest::new(path) {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                Err(CommandError::InvalidManifest(beautify_error_message(&e.to_string())))
+            }
+        }
+    }
+}
+
+fn beautify_error_message(error: &str) -> String {
+    let prefix = "parse error: ";
+
+    let mut stripped = error;
+
+    if let Some(s) = error.strip_prefix(prefix) {
+        stripped = s;
+    }
+
+    stripped.to_string()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+    use crate::commands::edge_app_settings::SettingType;
+
+    fn write_to_tempfile(dir: &tempfile::TempDir, file_name: &str, content: &str) -> std::path::PathBuf {
+        let file_path = dir.path().join(file_name);
+        std::fs::write(&file_path, content).unwrap();
+        file_path
+    }
+
+    fn serialize_deserialize_cycle(manifest: EdgeAppManifest) -> Result<EdgeAppManifest, CommandError> {
+        let dir = tempdir().unwrap();
+        let file_path = dir.path().join("screenly.yml");
+
+        EdgeAppManifest::save_to_file(&manifest, &file_path)?;
+        let new_manifest = EdgeAppManifest::new(&file_path)?;
+
+        Ok(new_manifest)
+    }
+
+    #[test]
+    fn test_save_manifest_to_file_should_save_yaml_correctly() {
+        let dir = tempdir().unwrap();
+        let file_path = dir.path().join("screenly.yml");
+
+        let manifest = EdgeAppManifest {
+            app_id: Some("test_app".to_string()),
+            user_version: Some("test_version".to_string()),
+            description: Some("test_description".to_string()),
+            icon: Some("test_icon".to_string()),
+            author: Some("test_author".to_string()),
+            homepage_url: Some("test_url".to_string()),
+            settings: vec![Setting {
+                title: "username".to_string(),
+                type_: SettingType::String,
+                default_value: "stranger".to_string(),
+                optional: true,
+                help_text: "An example of a setting that is used in index.html".to_string(),
+            }]
+        };
+
+        EdgeAppManifest::save_to_file(&manifest, &file_path).unwrap();
+
+        let contents = fs::read_to_string(file_path).unwrap();
+
+        let expected_contents = r#"---
```

Comment:
let's move this to a function, it repeats a lot

Reply:
expected contents are different every time, so I don't think we can put it into function